roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -2289,7 +2289,17 @@ RooDataSet *RooAbsPdf::generate(const RooArgSet &whatVars, const RooDataSet& pro
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+
+std::string RooAbsPdf::translate(RooFit::Detail::ADContextManager &ctxManager) const
+{
+   // For now just return function/normalization integral.
+   std::string func = ctxManager.nodeNames[servers()[0]->GetName()];
```

Comment:
```suggestion
   std::string func = ctxManager.getCode(servers()[0]);
```
You should abstract this away with a member function of the context for sevaral reasons:
1. You might change the data structure in the context in the future and it's not an `map` anymore
2. We might change the way how the name is translated into the lookup key later (for example, using the `namePtr()`.

Reply:
The `getCode` name was just an example by the way, feel free to come up with a better name!